Crc32c as submodule

KornelH opened the pull request:
Use `crc32c` as a Git sub-module instead of a build time dependency fetched by CMake. By this, we can pack all needed files and publish to Hex.pm and later on build it anywhere where all dependencies are already available (e.g. rebar3 and required hex packages) but there is no internet connection (e.g. isolated build environments).

zmstone commented:
Hi @KornelH 
Thank you for the PR, could you help to fix github action workflows

KornelH commented:
Of course, I'll check it.

KornelH committed:
Use latest `crc32c` version `2bbb3be`

KornelH committed:
Use `crc32c` as a submodule

Use `crc32c` as a Git sub-module instead of a build time dependency
fetched by CMake. By this, we can pack all needed files and publish to
Hex.pm and later on build it anywhere where all dependencies are already
available (e.g. rebar3 and required hex packages) but there is no
internet connection (e.g. isolated build environments).

KornelH committed:
Update git sub-modules in Workflows

zmstone commented on .github/workflows/main.yml:
```diff
@@ -28,6 +28,9 @@ jobs:
 
+    - name: Update git submodules
+      run: git submodule update --recursive --init
+
```
maybe update submodule in rebar3 pre-hook it's currently a git repo.
if not a git repo, then no submodule update (assuming the submodule dir is already downloaded from hex.pm).

KornelH replied:
I'm new in Gtihub Workflows, so I just tired to trigger it in my fork but it did not happened but triggered it in your repo by the pull request. Anyway, in the meantime I found that, the [action/checkout](https://github.com/actions/checkout) can be used to update sub-modules. I'll try that.

zmstone commented:
thanks @KornelH 
published 1.1.3 to hex.pm

and trying to make use of it here: https://github.com/kafka4beam/kafka_protocol/pull/148

KornelH commented:
I experienced a strange phenomenon that was not there when `crc32c` was a CMake dependency. It happens only when `crc32cer` is built in a [Bazel](https://bazel.build/) sandbox environment. When linking `libcrc32cer_nif.so`, the linker complains about `libcrc32c.a` cannot be relocated and it has to be recompiled with `-fPIC` (Position Independent Code). It is understandable as `libcrc32c.a` is a static library while `libcrc32cer_nif.so` is a shared library. I don't really know yet why it happens only in Bazel sandbox.

Anyway, I just turned on Position Independent Code for CMake targets of `crc32c` and it works perfectly. I just wanted to mention it here as it may be interesting.

```diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index a32b7f4..6aaf347 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -39,6 +39,9 @@ include_directories(${CMAKE_BINARY_DIR}/external/crc32c/include)
 
 # Specify the dependency on the external library
 add_dependencies(crc32cer_nif crc32c)
+set_target_properties(crc32c PROPERTIES POSITION_INDEPENDENT_CODE ON)
+set_target_properties(crc32c_arm64 PROPERTIES POSITION_INDEPENDENT_CODE ON)
+set_target_properties(crc32c_sse42 PROPERTIES POSITION_INDEPENDENT_CODE ON)
 set_property(TARGET crc32c PROPERTY IMPORTED_LOCATION ${CMAKE_BINARY_DIR}/external/src/google-crc32c-build/libcrc32c.a)
 
 if (CMAKE_SYSTEM_NAME MATCHES Darwin)
```
